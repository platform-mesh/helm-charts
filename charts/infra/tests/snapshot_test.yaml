suite: snapshot
values:
  - ../test-values.yaml
tests:
  - it: matches the snapshot
    asserts:
      - matchSnapshot: {}
  - it: disables externalsecrets
    set:
      externalSecrets:
        accountOperatorSaKubeconfig: null
    asserts:
      - matchSnapshot: {}
  - it: with https
    set:
      kcp:
        enabled: true
      istio:
        main:
          gateway:
            hosts:
              - "my-domain.com"
              - "my-domain2.com"
            port: 8000
            name: pass-https
            protocol: HTTPS
            tls:
              mode: SIMPLE
              credentialName: my-credential
              minProtocolVersion: TLSV1_2
    asserts:
      - matchSnapshot: {}
  - it: with kcp
    set:
      kcp:
        enabled: true
      istio:
        passThrough:
          gateway:
            enabled: true
            hosts:
              - "passthrough.my-domain.com"
              - "passthrough2.my-domain.com"
            port: 8000
            name: pass-https
            protocol: HTTPS

    asserts:
      - matchSnapshot: {}

