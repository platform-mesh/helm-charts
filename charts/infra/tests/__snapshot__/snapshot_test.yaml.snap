disables externalsecrets:
  1: |
    apiVersion: networking.istio.io/v1
    kind: Gateway
    metadata:
      name: gateway
      namespace: NAMESPACE
    spec:
      selector:
        istio: gateway
      servers:
        - hosts:
            - '*'
          port:
            name: http
            number: 8000
            protocol: HTTP
matches the snapshot:
  1: |
    apiVersion: networking.istio.io/v1
    kind: Gateway
    metadata:
      name: gateway
      namespace: NAMESPACE
    spec:
      selector:
        istio: gateway
      servers:
        - hosts:
            - '*'
          port:
            name: http
            number: 8000
            protocol: HTTP
with https:
  1: |
    apiVersion: networking.istio.io/v1
    kind: Gateway
    metadata:
      name: gateway
      namespace: NAMESPACE
    spec:
      selector:
        istio: gateway
      servers:
        - hosts:
            - my-domain.com
            - my-domain2.com
          port:
            name: pass-https
            number: 8000
            protocol: HTTPS
          tls:
            credentialName: my-credential
            minProtocolVersion: TLSV1_2
            mode: SIMPLE
  2: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: kcp-system-masters-client-cert-authz-webhook
      namespace: NAMESPACE
    spec:
      commonName: authz-webhook
      issuerRef:
        name: kcp-client-issuer
      privateKey:
        algorithm: RSA
        size: 2048
      secretName: kcp-system-masters-client-cert-authz-webhook
      subject:
        organizations:
          - system:masters
      usages:
        - client auth
  3: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: kcp-system-masters-client-cert-openmfp-operator
      namespace: NAMESPACE
    spec:
      commonName: openmfp-operator
      issuerRef:
        name: kcp-client-issuer
      privateKey:
        algorithm: RSA
        size: 2048
      secretName: kcp-system-masters-client-cert-openmfp-operator
      subject:
        organizations:
          - system:masters
      usages:
        - client auth
with kcp:
  1: |
    apiVersion: networking.istio.io/v1
    kind: Gateway
    metadata:
      name: gateway
      namespace: NAMESPACE
    spec:
      selector:
        istio: gateway
      servers:
        - hosts:
            - '*'
          port:
            name: http
            number: 8000
            protocol: HTTP
        - hosts:
            - passthrough.my-domain.com
            - passthrough2.my-domain.com
          port:
            name: pass-https
            number: 8000
            protocol: HTTPS
          tls:
            mode: PASSTHROUGH
  2: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: kcp-system-masters-client-cert-authz-webhook
      namespace: NAMESPACE
    spec:
      commonName: authz-webhook
      issuerRef:
        name: kcp-client-issuer
      privateKey:
        algorithm: RSA
        size: 2048
      secretName: kcp-system-masters-client-cert-authz-webhook
      subject:
        organizations:
          - system:masters
      usages:
        - client auth
  3: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      name: kcp-system-masters-client-cert-openmfp-operator
      namespace: NAMESPACE
    spec:
      commonName: openmfp-operator
      issuerRef:
        name: kcp-client-issuer
      privateKey:
        algorithm: RSA
        size: 2048
      secretName: kcp-system-masters-client-cert-openmfp-operator
      subject:
        organizations:
          - system:masters
      usages:
        - client auth
